common/authorization/oauthAuthorizer.go

```diff
@@ -107,19 +125,36 @@ func (a *oauthAuthority) parseToken(tokenStr string, verifier jwt.Verifier) (*jw
 	return &claims, nil
 }
 
-func (a *oauthAuthority) validateClaims(claims *jwtClaims, attributes *Attributes) error {
+func (a *oauthAuthority) validateTTL(claims *jwtClaims) error {
 	if claims.TTL > a.authorizationCfg.MaxJwtTTL {
 		return fmt.Errorf("TTL in token is larger than MaxTTL allowed")
 	}
 	if claims.Iat+claims.TTL < time.Now().Unix() {
 		return fmt.Errorf("JWT has expired")
 	}
-	if claims.Domain != attributes.DomainName {
-		return fmt.Errorf("domain in token doesn't match with current domain")
-	}
-	if NewPermission(claims.Permission) < attributes.Permission {
-		return fmt.Errorf("token doesn't have the right permission")
+	return nil
+}
+
+func (a *oauthAuthority) validatePermission(claims *jwtClaims, attributes *Attributes, data map[string]string) error {
+	groups := ""
+	switch attributes.Permission {
+	case PermissionRead:
+		groups = data[common.DomainDataKeyForReadGroups]
+	case PermissionWrite:
```

Comment:
Is write still write + read? We might want to change back the type of permissionwrite to string

Reply:
yeah you are right. Fixed it.

Reply:
Yes, you are right. Fixed it.